### ISSUE TITLE
[Feature idea] onLoaded event for Query component

### ISSUE THREAD

**Issue by javorosas** [apollographql/react-apollo#1927]
I have an use case that could justify the need for an onLoaded/onComplete event for the Query component. It goes like this:

I have a `select` input that loads its options asynchronously based on a query. When the query loads, I need to populate my `select`'s options and **select the first option by default**.

For now I can achieve this using the HOF and the (soon-to-be-deprecated) `componentWillReceiveProps`:

```
...
componentWillReceiveProps (props) {
  if (!props.loading && !props.error && props.data.options.length > 0 && !this.state.selected) {
    this.setState({ selected: props.data.options[0].value });
  }
}
...
```

I can't see how this can be achieved using the Query component without having an event which fires up when the query has been fetched (similar to the one the Mutation component has). Any ideas?

**Comment by javorosas** [comment]
Didn't notice this is duplicated from #1862 and being addressed in #1922. Closing.

**Comment by tab00** [comment]
We need onLoaded in Query. What's going on?
